src/items.rs

```diff
@@ -1462,9 +1471,15 @@ fn rewrite_where_clause(context: &RewriteContext,
     };
     let preds_str = try_opt!(write_list(&item_vec, &fmt));
 
-    // 9 = " where ".len() + " {".len()
+    // When '{' is the terminator just assume that it is put on the next line for single line where
+    // clauses
```

Comment:
Is this ever not the case? I'd rather we are sub-optimal by one char than even occasionally over-shoot by a char.